plugin/nfs/nfs_clib.c

```diff
@@ -0,0 +1,166 @@
+#include <Python.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <sys/statvfs.h>
+#include <mntent.h>
+
+#include <libstoragemgmt/libstoragemgmt.h>
+
+
+#define _BUFF_SIZE   1024
+#define LSF_LOCAL_MOUNTS "/proc/self/mounts"
+
+
+
+#if PY_MAJOR_VERSION > 2
+#define PyInt_FromLong PyLong_FromLong
+#endif
+
+#if PY_MAJOR_VERSION >= 3
+    #define MOD_DEF(name, methods) \
+        static struct PyModuleDef moduledef = { \
+            PyModuleDef_HEAD_INIT, name, NULL, -1, methods, NULL, NULL, NULL, \
+            NULL}; \
+        return PyModule_Create(&moduledef);
+#else
+    #define MOD_DEF(name, methods) \
+        Py_InitModule(name, methods);
+#endif
+
+#define _UNUSED(x) (void)(x)
+
+
+static PyObject * find_fsid_bypath(PyObject * self, PyObject *args)
+{
+    const char * path = NULL;
+    PyObject * err = NULL;
+    char * out = NULL;
+    PyObject * result = NULL;
+
+    _UNUSED(self);
+
+    if (!PyArg_ParseTuple(args, "s", &path))
+        return NULL;
+
+    struct statvfs st;
+
+    if (statvfs(path, &st)==-1) {
+        err = PyExc_OSError;
+        PyErr_SetFromErrno(err);
+        return NULL;
+    }
+
+    if (st.f_fsid == 0) {
+        err = PyExc_OSError;
+        PyErr_SetString(err, "No FSID found");
+        return NULL;
+    }
+
+    if (asprintf(&out, "%lx", st.f_fsid)<0) {
```

Comment:
Already defined by including Python.h, to define it again is an error, can add a conditional define of it but its somewhat redundant.